sdks/java/harness/src/main/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClient.java

```diff
@@ -102,7 +104,7 @@ public BeamFnLoggingClient(
       Function<Endpoints.ApiServiceDescriptor, ManagedChannel> channelFactory) {
     this.apiServiceDescriptor = apiServiceDescriptor;
     this.inboundObserverCompletion = new CompletableFuture<>();
-    this.phaser = new Phaser(1);
```

Comment:
Since this was a normal Phaser, I think that it didn't block for flow control previously after the first OnReady false->True transition.